File: backend/app/models/flow.py
from enum import Enum

from pydantic import BaseModel


class PhaseType(str, Enum):
    NEUTRAL = "neutral"
    POSE_HOLD = "pose_hold"
    HAND_MOTION = "hand_motion"


class TrackedPoint(str, Enum):
    HEAD = "head"
    LEFT_SHOULDER = "left_shoulder"
    RIGHT_SHOULDER = "right_shoulder"
    LEFT_ELBOW = "left_elbow"
    RIGHT_ELBOW = "right_elbow"
    LEFT_HAND = "left_hand"
    RIGHT_HAND = "right_hand"


class Phase(BaseModel):
    index: int
    name: str
    phase_type: PhaseType
    start_sec: float
    end_sec: float
    tracked_points: list[TrackedPoint]
    description: str


class Flow(BaseModel):
    flow_id: str
    name: str
    duration_sec: float
    phases: list[Phase]

File: frontend/types/flow.ts
export type PhaseType = "neutral" | "pose_hold" | "hand_motion";

export type TrackedPoint =
  | "head"
  | "left_shoulder"
  | "right_shoulder"
  | "left_elbow"
  | "right_elbow"
  | "left_hand"
  | "right_hand";

export interface Phase {
  index: number;
  name: string;
  phase_type: PhaseType;
  start_sec: number;
  end_sec: number;
  tracked_points: TrackedPoint[];
  description: string;
}

export interface Flow {
  flow_id: string;
  name: string;
  duration_sec: number;
  phases: Phase[];
}
